target/linux/ath79/image/gen_kt_img.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+SUFFIX=.patched
+TMP_FILE1="/tmp/__tmp_file1__"
+TMP_FILE2="/tmp/__tmp_file2__"
+TMP_FILE3="/tmp/__tmp_file3__"
+
+function usage() {
+  echo ""
+  echo "$0 (bin file)"
+  echo ""
+}
+
+function patch_uimage_header() {
+  if [ ! -f ${1} ]
+  then
+    echo "File not found !"
+    exit 1
+  fi
+
+  dd if=${1} of=${TMP_FILE1} count=64 bs=1
```

Comment:
I don't get why you create so many intermediate files. This should all be possible like this:

```
dd if=$1 of=$1.tmp count=1 bs=4
printf '\x00\x00\x00\x00\x00\x02\x04\x00' | dd of=$1.tmp bs=1 seek=4 conv=notrunc
dd bs=1 skip=8 if=$1 >> $1.tmp

CRC32_CHECKSUM=$(crc32 $1.tmp)
echo "*** CRC32 Checksum: ${CRC32_CHECKSUM}"

echo "${CRC32_CHECKSUM}" | xxd -r -p | dd of=$1.tmp bs=1 seek=4 conv=notrunc

dd bs=64 skip=1 if=$1 of=$1.tmp oflag=append conv=notrunc

mv $1.tmp $1
```

There is still room for improvement here, i.e. dd vs. >> etc.